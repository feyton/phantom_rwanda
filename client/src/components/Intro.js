import React from 'react';
import { useTranslation } from 'react-i18next';
import { Link } from 'react-router-dom';
import bus from '../assets/bus.png';
import Button from './Button.js';

const Intro = () => {
  const { t } = useTranslation();
  return (
    <div className="flex flex-col items-center lg:flex-row xl:flex-row">
      <div>
        <h1 className="flex flex-col text-3xl md:text-5xl xl:text-5xl font-raleway font-extrabold mx-12 xl:mx-28 mt-24 md:mt-32 xl:mt-32">
          <span>{t('hero_head_1')}</span>
          <span className="md:ml-10 ml-4">{t('hero_head_2')}</span>
        </h1>
        <div className="flex flex-col text-sm md:text-xl xl:text-base mx-20 md:ml-32 xl:ml-52 mt-10">
          <p>{t('hero_text_1')}</p>
          <div>
            <p className="ml-4">{t('hero_text_2')}</p>
          </div>
        </div>
        <div className="flex flex-row justify-center w-fit">
          <Link to="tracking-page">
            <Button
              name={t('hero_btn')}
              styles="bg-primary hover:bg-hover text-white rounded-3xl mt-12 px-12 xl:px-12 py-2 ml-18 md:ml-28 xl:ml-44"
            />
          </Link>
        </div>
      </div>
      <div className="w-full px-20 lg:w-80 xl:w-96 lg:ml-auto xl:ml-auto lg:mr-36 xl:mr-36 lg:px-0 xl:px-0 mt-28">
        <img src={bus} alt="Bus" />
      </div>
    </div>
  );
};

export default Intro;
