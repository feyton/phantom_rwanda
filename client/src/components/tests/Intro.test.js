import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { shallow } from 'enzyme';
import Intro from '../Intro.js';

describe('Intro', () => {
  it('contains one child elements', () => {
    const component = shallow(<Intro />);
    const value = component.children().length;
    expect(value).toEqual(2);
  });
  it('should render Intro', () => {
    const intro = renderer
      .create(
        <MemoryRouter>
          <Intro />
        </MemoryRouter>
      )
      .toJSON();
    expect(intro).toMatchSnapshot();
  });
});
