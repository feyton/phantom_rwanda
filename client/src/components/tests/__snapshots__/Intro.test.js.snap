// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Intro should render Intro 1`] = `
<div
  className="flex flex-col items-center lg:flex-row xl:flex-row"
>
  <div>
    <h1
      className="flex flex-col text-3xl md:text-5xl xl:text-5xl font-raleway font-extrabold mx-12 xl:mx-28 mt-24 md:mt-32 xl:mt-32"
    >
      <span>
        hero_head_1
      </span>
      <span
        className="md:ml-10 ml-4"
      >
        hero_head_2
      </span>
    </h1>
    <div
      className="flex flex-col text-sm md:text-xl xl:text-base mx-20 md:ml-32 xl:ml-52 mt-10"
    >
      <p>
        hero_text_1
      </p>
      <div>
        <p
          className="ml-4"
        >
          hero_text_2
        </p>
      </div>
    </div>
    <a
      href="/tracking-page"
      onClick={[Function]}
    >
      <div>
        <button
          className="bg-primary hover:bg-hover text-white rounded-3xl mt-12 px-12 xl:px-12 py-2 ml-28 md:ml-48 xl:ml-64 rounded-xl py-2 text-base md:text-sm px-5"
          type="submit"
        >
          hero_btn
        </button>
      </div>
    </a>
  </div>
  <div
    className="w-full px-20 lg:w-80 xl:w-96 lg:ml-auto xl:ml-auto lg:mr-36 xl:mr-36 lg:px-0 xl:px-0 mt-28"
  >
    <img
      alt="Bus"
      src="bus.png"
    />
  </div>
</div>
`;
